src/cmakeProject.ts

```diff
@@ -266,11 +266,24 @@ export class CMakeProject {
             return undefined;
         }
         log.debug(localize('resolving.config.preset', 'Resolving the selected configure preset'));
+
+        // We want to use the original unexpanded preset file to apply the dev env in expandConfigurePreset
+        // we have to first check if the preset is valid in expandedPresetsFiles since we won't be expanding the whole file here, only the path up for this preset
```

Comment:
I noticed a very similar comment and code to this inside of the `expandConfigurePreset` method. Do we need both? It feels like it should only be in one place, likely here?